pkgs/by-name/fo/foxglove-studio/package.nix

```
@@ -0,0 +1,109 @@
+{
+  alsa-lib,
+  at-spi2-atk,
+  autoPatchelfHook,
+  cairo,
+  cups,
+  dbus,
+  dpkg,
+  eudev,
+  expat,
+  fetchurl,
+  glib,
+  glibc,
+  gtk3,
+  lib,
+  libdrm,
+  libgcc,
+  libxkbcommon,
+  makeWrapper,
+  mesa,
+  nspr,
+  nss,
+  pango,
+  stdenv,
+  systemd,
+  via,
+  xorg,
+}:
+stdenv.mkDerivation (finalAttrs: {
+  pname = "foxglove-studio";
+  version = "2.18.1";
+
+  src = fetchurl {
+    url = "https://get.foxglove.dev/desktop/v${finalAttrs.version}/foxglove-studio-${finalAttrs.version}-linux-amd64.deb";
+    sha256 = "sha256-vHloPOmrr8HR9Su5s84xohh8uCpbG4D9eEqbp+jPsDc=";
+  };
+
+  nativeBuildInputs = [
+    autoPatchelfHook
+    dpkg
+    makeWrapper
+  ];
+
+  buildInputs = [
+    alsa-lib
+    at-spi2-atk
+    cairo
+    cups
+    dbus
+    eudev
+    expat
+    glib
+    glibc
+    gtk3
+    libdrm
+    libgcc.libgcc
+    libxkbcommon
+    mesa
+    nspr
+    nss
+    pango
+    systemd
+    via
+    xorg.libX11
+    xorg.libXcomposite
+    xorg.libXdamage
+    xorg.libXext
+    xorg.libXfixes
+    xorg.libXrandr
+    xorg.libxcb
+  ];
+
+  preFixup = ''
+    addAutoPatchelfSearchPath "$out/opt/Foxglove Studio/"
+  '';
+
+  unpackPhase = ''
+    dpkg-deb -x $src .
+  '';
+
+  installPhase = ''
+    runHook preInstall
+
+    mkdir -p $out/bin
+    cp -R usr/share opt $out/
+
+    # fix the path in the desktop file
+    substituteInPlace \
+      $out/share/applications/foxglove-studio.desktop \
+      --replace /opt/ $out/opt/
```

```suggestion
      --replace /opt/.... "foxglove-studio"
```
Usually we avoid absolute paths in .desktop files #308324